### PR TITLE
Rust 1.62.0

### PR DESCRIPTION
**Pull request by ojeda**
Signed-off-by: Miguel Ojeda <ojeda@kernel.org>

### CODE REVIEW

**Comment by alex** [comment]
I skipped the `alloc/` dir, but rest of the changes LGTM.

**Comment by ojeda** [comment]
Thanks Alex for the quick review! There were a few changes in our own `alloc` bits to match the updates done to the originals, as usual, but nothing major.

File: rust/compiler_builtins.rs [hunk 1]
```diff
@@ -36,6 +36,18 @@ macro_rules! define_panicking_intrinsics(
     }
 );
 
+define_panicking_intrinsics!("`f32` should not be used", {
```
Comment (ojeda) [inline rust/compiler_builtins.rs:39]: @nbdd0121 @bjorn3 We got a few new intrinsics generated from `f{32,64}::classify`

Comment (bjorn3) [reply rust/compiler_builtins.rs:39]: Looks like someone forgot `#[inline]` on `classify`.

**Review by bjorn3 — APPROVED** [review]

**Pull request merged**